### PR TITLE
Feat/pism on albedo

### PR DESCRIPTION
**Pull request by ackerlar**
Branch that includes PISM and FESOM2.5 on albedo and reorganizes the iterative awiesm pism coupling. The coupling scripts are no longer in `runscripts/awicm-pism/coupling/` but in `couplings/`. Hence, `script_dir` in old awiesm-pism runscripts needs to be updated. 

Example runscripts for the coupling are in `runscripts/awicm-pism/` (`awiesm_pism_cpl-albedo-initial-yearly.yaml`, `pism_cpl-albedo-initial-yearly.yaml`, `awiesm_cpl-albedo-initial-yearly.yaml`)

An example runscript for FESOM2.5 with icebergs is in `runscripts/awiesm/awiesm-2.1_icebergs_albedo.yaml`

### CODE REVIEW

File: configs/components/echam/echam.yaml
Comment (mandresm) [inline configs/components/echam/echam.yaml]: Is this bound to the version of echam? I'm asking because if it is not, other versions of echam might break when you add this new variables to the namelist.

File: configs/components/echam/echam.yaml
Comment (mandresm) [inline configs/components/echam/echam.yaml]: this can always be in the general `input_sources`, to reduce the amount of verbosity. Only `input_files` decide which `input_files` from the available sources go into the simulation.

File: configs/components/echam/echam.yaml
Comment (mandresm) [inline configs/components/echam/echam.yaml]: this can always be in the general `input_sources`, to reduce the amount of verbosity. Only `input_files` decide which `input_files` from the available sources go into the simulation.

File: configs/components/echam/echam.yaml
Comment (mandresm) [inline configs/components/echam/echam.yaml]: Same as above

File: configs/components/fesom/fesom-2.1.yaml
Comment (mandresm) [inline configs/components/fesom/fesom-2.1.yaml]: This is specific to version 2.5 correct? then it needs to go under the version choose_block

File: configs/components/fesom/fesom-2.1.yaml
Comment (mandresm) [inline configs/components/fesom/fesom-2.1.yaml]: can I have that new line back please? :) less differences is better for us, the poor maintainers :)

File: configs/components/fesom/fesom-2.1.yaml
Comment (mandresm) [inline configs/components/fesom/fesom-2.1.yaml]: can I have that new line back?

File: configs/components/fesom/fesom-2.1.yaml
Comment (mandresm) [inline configs/components/fesom/fesom-2.1.yaml]: Remove these comments if it works
```suggestion
        icb_restart: iceberg.restart
        icb_restart_ISM: iceberg.restart.ISM
```

File: configs/components/fesom/fesom-2.1.yaml
Comment (mandresm) [inline configs/components/fesom/fesom-2.1.yaml]: Remove the comments if it works

```suggestion
        icb_restart: iceberg.restart.${end_date!syear}
        icb_restart_ISM: iceberg.restart.ISM.${end_date!syear}
```

File: configs/components/fesom/fesom-2.1.yaml
Comment (mandresm) [inline configs/components/fesom/fesom-2.1.yaml]: This is tricky, this will change the behaviour of FESOM output writting for everyone... such a change could be a pain, specially for people running monthly and daily simulations. I understand we are renaming the output from FESOM... not an easy question to answer, given that many many users have a say. I would say, you guys can make a decision about this in Paleo, and then you can put it as part of AWIESM.yaml so that it doesn't conflict with other people using fesom.yaml? or even in the runscript? The other option would be to change it for everyone here, as you are suggesting.

File: configs/components/fesom/fesom-2.1.yaml
Comment (mandresm) [inline configs/components/fesom/fesom-2.1.yaml]: why would want to remove the `add_` here?

File: configs/components/fesom/fesom-2.1.yaml
Comment (mandresm) [inline configs/components/fesom/fesom-2.1.yaml]: you can remove these lines since they are outside of the choose block already :)

File: configs/components/fesom/fesom-2.1.yaml
Comment (mandresm) [inline configs/components/fesom/fesom-2.1.yaml]: do you need an `add_` here?

File: configs/components/fesom/fesom-2.1.yaml
Comment (mandresm) [inline configs/components/fesom/fesom-2.1.yaml]: There is a lot of repetition here. Have you considered using the multilist (list loops) feature?

https://esm-tools.readthedocs.io/en/latest/yaml.html#list-loops

Let me know if you need help with that.

File: configs/components/fesom/fesom-2.1.yaml
Comment (mandresm) [inline configs/components/fesom/fesom-2.1.yaml]: Relevant only for version 2.5. Can we place them under the choose block for the version?

**Review by mandresm — CHANGES_REQUESTED** [review]
I have reviewed until the `fesom-2.1.yaml`. Good job, I'll keep reviewing tomorrow.

File: configs/components/fesom/fesom-2.1.yaml
Comment (ackerlar) [reply configs/components/fesom/fesom-2.1.yaml]: Does it make sense to have a new fesom config file "fesom-2.5.yaml" additional to "fesom-2.1.yaml"? Or is the idea to  put all versions into a general "fesom.yaml"?

**Comment by mandresm** [comment]
Hey @ackerlar! Is it possible to avoid including a version that is called `refact`? `reffactoring` branch in FESOM is a development branch, and I would like to keep it away from ESM-Tools yamls as much as possible, because it will be a pain for us to maintain configurations there.

If the branch of fesom used by you is not `refact` but a more stable branch/tag under your control, or even tag 2.5 then please give it a descriptive name (2.5, awiesm-pism or something similar) so that it doesn't lead to the AWIESM users to the confusion/illusion that they are using FESOM-Refactoring branch.

**Comment by ackerlar** [comment]
Hi @mandresm, I changed the name from `refac` to `paleodyn-2.5`

File: src/esm_runscripts/cli.py
Comment (mandresm) [inline src/esm_runscripts/cli.py]: Why did you needed to remove this line?

File: src/esm_runscripts/namelists.py
Comment (mandresm) [inline src/esm_runscripts/namelists.py:349]: Just out of curiosity, why did you need the `header=None`?

File: src/esm_runscripts/cli.py
Comment (ackerlar) [reply src/esm_runscripts/cli.py]: To be honest, I don't remember... I am currently testing if it affects my runs

File: src/esm_runscripts/namelists.py
Comment (ackerlar) [reply src/esm_runscripts/namelists.py:349]: Just out of convenience for me. In my opinion, it is less ambiguous, but I am totally fine with not setting `header`. I don't know how many people are using this though

Comment (pgierz) [reply src/esm_runscripts/namelists.py:349]: I at some point wrote the transient forcing table function. `header=None` is just something you give to pandas to ensure that it does not use the first row as a label for the columns. The default is "infer" (https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html), and explicitly setting `None` seems to be safer (in my opinion)

Comment (mandresm) [reply src/esm_runscripts/namelists.py:349]: thanks for the clarifications guys!

File: configs/components/fesom/fesom-2.1.yaml
Comment (mandresm) [reply configs/components/fesom/fesom-2.1.yaml]: One day we could consider (in a discussion with all of you model devs) putting everything into fesom-2.yaml. For now let's keep the minor versions separated.

File: .gitignore
Comment (mandresm) [inline .gitignore:154]: I am not sure about this change, @pgierz do you have an opinion about it?

File: configs/components/echam/echam.yaml
Comment (mandresm) [inline configs/components/echam/echam.yaml:25]: Waiting from a review in https://gitlab.awi.de/paleodyn/Models/echam6/-/merge_requests/5

**Review by mandresm — COMMENTED** [review]
Some comments, I still need to keep reviewing but that's something for tomorrow :)

File: configs/components/jsbach/jsbach.yaml
Comment (mandresm) [inline configs/components/jsbach/jsbach.yaml:357]: This will probably break restarts for every old simulation. I think we need to revert this changes

File: configs/components/jsbach/jsbach.yaml
Comment (mandresm) [inline configs/components/jsbach/jsbach.yaml]: What do the `jsbach: jsbach` lines do?

File: configs/setups/awiesm/awiesm.yaml
Comment (mandresm) [inline configs/setups/awiesm/awiesm.yaml:92]: Why not `2.1.1`? You can add a comment on top explaining what versions of FESOM are used in each AWIESM version if more clarity is needed.

File: configs/setups/awiesm/awiesm.yaml
Comment (mandresm) [inline configs/setups/awiesm/awiesm.yaml:95]: Same as above, why not `2.1.1`? I am fine with any versioning name, is totally up to you. I personally prefer Jan's approach on adding the minor numbers because it makes things less complicated and it gives a sense of what is the latest version.

File: configs/setups/awiesm/awiesm.yaml
Comment (mandresm) [inline configs/setups/awiesm/awiesm.yaml:192]: If you decide to change the version name, you'll have to also change it here.

Comment (mandresm) [reply configs/setups/awiesm/awiesm.yaml:192]: Actually, you could unify things quite a few things with a new variable `major_version`. If you are interested on having a look at that please let me know, we can do it on a one-to-one meeting today.

File: configs/setups/awiesm/awiesm.yaml
Comment (mandresm) [inline configs/setups/awiesm/awiesm.yaml]: This is the question you had about dynamic_vegetations right? I have no idea why I suggested to remove "${dynamic_vegetations}". You can put it back and we will see if it kills something in the tests. If it doesn't it can stay.

File: configs/setups/awiesm/awiesm.yaml
Comment (mandresm) [inline configs/setups/awiesm/awiesm.yaml:320]: Same as above, if you are changing the version name, you'll also have to change it here.

File: setup.py
Comment (mandresm) [inline setup.py]:
```suggestion
```

Now that those are installed in the plugin, this can be removed

File: src/esm_parser/esm_parser.py
Comment (mandresm) [inline src/esm_parser/esm_parser.py]:
```suggestion
```

Remove this change.

**Review by mandresm — CHANGES_REQUESTED** [review]
Final round of suggestions/reviews

File: configs/components/jsbach/jsbach.yaml
Comment (mandresm) [reply configs/components/jsbach/jsbach.yaml:357]: bReaks backcompatibility with simulation restarts with_wiso. I don't think that's a big deal, since not so much people is using with_wiso yet in production right @ackerlar?

**Comment by mandresm** [comment]
@ackerlar, all okay from my side, you can go ahead and do the last run of tests to see if awiesm-pism is still runing. Let me know how that goes :)

**Review by mandresm — APPROVED** [review]

**Comment by mandresm** [comment]
Do not merge yet, missing the last steps on the testing approvals, and the final test from Lars to check that everything is still working on AWIESM PISM and icebergs

**Comment by ackerlar** [comment]
I tested icebergs and pism coupling, and both works fine for me! :)

**Comment by mandresm** [comment]
Let's go! #approve-changes :D

**Comment by mandresm** [comment]
#bump #minor

**Pull request merged**
